kytos/core/auth.py

```diff
@@ -0,0 +1,334 @@
+            del answer['data']['password']
+        return answer, code
+
+    @authenticated
+    def _list_users(self):
+        """List all users using Storehouse."""
+        response = {}
+
+        def _list_users_callback(_event, boxes, error):
+            nonlocal response
+            if error or boxes is None:
+                response = {
+                    "answer": "Users cannot be listed",
+                    "code": HTTPStatus.INTERNAL_SERVER_ERROR.value,
+                }
+            else:
+                response = {
+                    "answer": {"users": boxes},
+                    "code": HTTPStatus.OK.value,
+                }
+
+        content = {
+            "namespace": self.namespace,
+            "callback": _list_users_callback,
+        }
+        event = KytosEvent(name="kytos.storehouse.list", content=content)
+        self.controller.buffers.app.put(event)
+        while True:
+            time.sleep(0.1)
+            if response:
+                break
+        return response["answer"], response["code"]
+
+    @authenticated
+    def _create_user(self):
+        """Save a user using Storehouse."""
+        response = {}
+
+        def _create_user_callback(_event, box, error):
+            nonlocal response
+            if error or not box:
+                response = {
+                    "answer": "User cannot be created",
+                    "code": HTTPStatus.CONFLICT.value,
+                }
+            else:
+                response = {
+                    "answer": "User successfully created",
+                    "code": HTTPStatus.OK.value,
+                }
+
+        req = request.json
+        password = req["password"].encode()
+        data = {
+            "username": req["username"],
+            "email": req["email"],
+            "password": hashlib.sha512(password).hexdigest(),
+        }
+        content = {
+            "namespace": self.namespace,
+            "box_id": data["username"],
+            "data": data,
+            "callback": _create_user_callback,
+        }
+        event = KytosEvent(name="kytos.storehouse.create", content=content)
+        self.controller.buffers.app.put(event)
+        while True:
+            time.sleep(0.1)
+            if response:
+                break
+        return response["answer"], response["code"]
+
+    @authenticated
+    def _delete_user(self, uid):
+        """Delete a user using Storehouse."""
+        response = {}
+
+        def _delete_user_callback(_event, box, error):
+            nonlocal response
+            if error or not box:
+                response = {
+                    "answer": "User has not been deleted",
+                    "code": HTTPStatus.INTERNAL_SERVER_ERROR.value,
+                }
+            else:
+                response = {
+                    "answer": "User successfully deleted",
+                    "code": HTTPStatus.OK.value,
+                }
```

Comment:
Same as `_find_user`. Here maybe it would be better to check first if the user exists.